xds/internal/xdsclient/bootstrap/bootstrap.go

```diff
@@ -48,18 +50,50 @@ const (
 	// server supports the v3 version of the xDS transport protocol.
 	serverFeaturesV3 = "xds_v3"
 
-	// Type name for Google default credentials.
-	credsGoogleDefault              = "google_default"
-	credsInsecure                   = "insecure"
 	gRPCUserAgentName               = "gRPC Go"
 	clientFeatureNoOverprovisioning = "envoy.lb.does_not_support_overprovisioning"
 )
 
+func init() {
+	bootstrap.RegisterCredentials(&insecureCredsBuilder{})
+	bootstrap.RegisterCredentials(&googleDefaultCredsBuilder{})
+}
+
 var gRPCVersion = fmt.Sprintf("%s %s", gRPCUserAgentName, grpc.Version)
 
 // For overriding in unit tests.
 var bootstrapFileReadFunc = ioutil.ReadFile
 
+// insecureCredsBuilder encapsulates a insecure credential that is built using a
+// JSON config.
+type insecureCredsBuilder struct{}
+
+// Build returns a default insecure credential bundle. Currently the JSON
+// config is unused.
```

Comment:
You don't need to document this at all, since the type is unexported and it exists to satisfy an interface, plus the function is trivial and self-documenting.  Same with `Name()` and the methods on `googleDefaultCredsBuilder`.

(Generally: if a comment has no helpful information in it, it just adds clutter, so remove it.)

Reply:
Removed.